astroquery/mast/cloud.py

```diff
@@ -109,30 +109,13 @@ def get_cloud_uri(self, data_product, include_bucket=True, full_url=False):
             found in the cloud, None is returned.
         """
 
-        s3_client = self.boto3.client('s3', config=self.config)
-
-        path = utils.mast_relative_path(data_product["dataURI"])
-        if path is None:
-            raise InvalidQueryError("Malformed data uri {}".format(data_product['dataURI']))
+        uri_list = self.get_cloud_uri_list(data_product, include_bucket=include_bucket, full_url=full_url)
 
-        if 'galex' in path:
-            path = path.lstrip("/mast/")
+        # Making sure we got at least 1 URI from the query above.
+        if len(uri_list) == 0:
+            raise InvalidQueryError("No products found on the cloud for this query.")
         else:
-            path = path.lstrip("/")
-
-        try:
-            s3_client.head_object(Bucket=self.pubdata_bucket, Key=path)
-            if include_bucket:
-                path = "s3://{}/{}".format(self.pubdata_bucket, path)
-            elif full_url:
-                path = "http://s3.amazonaws.com/{}/{}".format(self.pubdata_bucket, path)
-            return path
-        except self.botocore.exceptions.ClientError as e:
-            if e.response['Error']['Code'] != "404":
-                raise
-
-        warnings.warn("Unable to locate file {}.".format(data_product['productFilename']), NoResultsWarning)
-        return None
+            return uri_list[0]
```

Comment:
Why only the first element, what is expected to be included in the rest?

Reply:
The output for `get_cloud_uri_list` is always a list, even when the output returns only 1 URI, so the sole element gets indexed out before being returned. I just committed some commentary above this line for clarity.

Reply:
There wouldn't be a case where a request for a single cloud URI would return more than one because of L171 in `util.py` of this branch, but I can raise an AssertionError here to ensure that there is only 1 element in the list, what do you think?

Reply:
I'm really going into bikeshedding now, but if that's the case, why does it have to be a list to begin with?

Also, L173 is never reached? 

Maybe just a comment about this would be enough, no need for the assertion.